packages/arb-rpc-node/aggregator/aggregator.go

```diff
@@ -161,22 +165,35 @@ func (m *Server) LatestSnapshot(ctx context.Context) (*snapshot.Snapshot, error)
 }
 
 func (m *Server) PendingSnapshot(ctx context.Context) (*snapshot.Snapshot, error) {
-	pending, err := m.batch.PendingSnapshot(ctx)
-	if err != nil {
-		return nil, err
-	}
-	if pending == nil {
-		return m.LatestSnapshot(ctx)
+	if m.batch != nil {
+		pending, err := m.batch.PendingSnapshot(ctx)
+		if err != nil {
+			return nil, err
+		}
+		if pending == nil {
+			return m.LatestSnapshot(ctx)
+		}
+		return pending, nil
 	}
-	return pending, nil
+
+	return nil, errors.New("no batcher defined, cannot generate pending snapshot")
```

Comment:
We should return `m.LatestSnapshot(ctx)` in this case

Reply:
:+1: